src/main/java/org/apache/solr/benchmarks/BenchmarksMain.java

```diff
@@ -178,49 +260,52 @@ public static void runIndexingBenchmarks(List<IndexBenchmark> indexBenchmarks, S
 		}
 	}
 
-    private static Supplier<ControlledExecutor.CallableWithType<String, NamedList<Object>>> getQuerySupplier(QueryGenerator queryGenerator, HttpSolrClient client, String collection) {
-		class QueryCallable implements ControlledExecutor.CallableWithType<String, NamedList<Object>> {
-			private final QueryRequest queryRequest;
+	public static class QueryCallable implements ControlledExecutor.CallableWithType<String, SolrBenchQueryResponse> {
+		private final QueryRequest queryRequest;
+		private final String queryString;
+		private final HttpSolrClient client;
+		private final String collection;
+		
+		public QueryCallable(QueryRequest queryRequest, String queryString, HttpSolrClient client, String collection) {
+			this.queryRequest = queryRequest;
+			this.client = client;
+			this.collection = collection;
+			this.queryString = queryString;
```

Comment:
Is this different from the `queryRequest.toString()` ? getType() and getQueryString() might return very similar result. `getType()` might not be the best naming that i used 😓  but it was actually used to identify stats for different "types" which each type is just the query.